openwisp_monitoring/device/schema.py

```diff
@@ -145,6 +145,42 @@
                             "tx_carrier_errors": {"type": "integer"},
                         },
                     },
+                    "wireless": {
+                        "type": "object",
+                        "properties": {
+                            "channel": {"type": "integer"},
+                            "country": {"type": "string"},
+                            "frequency": {"type": "integer"},
+                            "mode": {"type": "string"},
+                            "noise": {"type": "integer"},
+                            "signal": {"type": "integer"},
+                            "ssid": {"type": "string"},
+                            "tx_power": {"type": "integer"},
+                            "clients": {
+                                "type": "array",
+                                "title": "Wireless Clients",
+                                "additionalItems": True,
+                                "items": {
+                                    "type": "object",
+                                    "title": "Client",
+                                    "properties": {
+                                        "aid": {"type": "integer"},
+                                        "assoc": {"type": "boolean"},
+                                        "auth": {"type": "boolean"},
+                                        "authorized": {"type": "boolean"},
+                                        "ht": {"type": "boolean"},
+                                        "vht": {"type": "boolean"},
+                                        "wds": {"type": "boolean"},
+                                        "wmm": {"type": "boolean"},
+                                        "wps": {"type": "boolean"},
+                                        "mac": {"type": "string"},
+                                        "mfp": {"type": "boolean"},
+                                        "preauth": {"type": "boolean"},
```

Comment:
Please add also:

* `additionalProperties` false
* `additionalItems` -> change to false
* `required: [aid, assoc, auth... all of them]`

Reply:
You have not added additionalProperties and additionalItems to the correct places.

`additionalProperties` has to be in the same level as `properties` and similarly `additionalItems` for `items`.

I think reading carefully the rest of the schema and following this guide will help you understand it better :+1:

https://json-schema.org/learn/getting-started-step-by-step.html